compiler/noirc_evaluator/src/ssa/opt/array_set.rs

```diff
@@ -55,9 +58,11 @@ impl Function {
 fn analyze_last_uses(
     dfg: &DataFlowGraph,
     block_id: BasicBlockId,
+    is_brillig_func: bool,
     array_to_last_use: &mut HashMap<ValueId, InstructionId>,
     instructions_that_can_be_made_mutable: &mut HashSet<InstructionId>,
     arrays_from_load: &mut HashSet<ValueId>,
+    inner_nested_arrays: &mut HashMap<ValueId, InstructionId>,
```

Comment:
This method is getting quite long. I am going to make a context struct in a follow-up refactor.